codegen/post_gen_hooks.go

```diff
@@ -330,3 +341,162 @@ func writeAndFormat(path string, data []byte) error {
 	}
 	return FormatGoFile(path)
 }
+
+// WorkflowMockGenHook returns a PostGenHook to generate endpoint workflow mocks
+func WorkflowMockGenHook(h *PackageHelper, t *Template) PostGenHook {
+	return func(instances map[string][]*ModuleInstance) error {
+		errChanSize := 0
+		shouldGenMap := map[*ModuleInstance][]*EndpointSpec{}
+		for _, instance := range instances["endpoint"] {
+			endpointSpecs := instance.genSpec.([]*EndpointSpec)
+			for _, endpointSpec := range endpointSpecs {
+				if endpointSpec.WorkflowType == "custom" {
+					shouldGenMap[instance] = endpointSpecs
+					errChanSize += len(endpointSpecs)
```

Comment:
this `+=` makes me think there're more custom endpoint specs and we'll accumulate, but it follows with a `break` :)

Reply:
hmm, I don't get it